device/iot-device-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsDeviceTelemetryTest.java

```diff
@@ -544,6 +549,15 @@ public void protonMessageToIoTHubMessageWithInputName(@Mocked final MessageImpl
                 mockMessage.getMessageAnnotations();
                 result = mockedAnnotations;
 
+                mockMessage.getProperties();
+                result = mockProperties;
+
+                mockProperties.getUserId();
+                result = mockBinary;
+
+                mockBinary.toString();
+                result = "someUserId";
```

Comment:
The userId wasn't being mocked properly, and it was hitting the codepath in MessageProperty.java that complains about the contents of the userId not being valid. Without this change, it was using a memory address with several illegal characters